test/NNPDE_tests.jl

```diff
@@ -19,30 +26,45 @@ bcs = [u(0.,θ) ~ 1.0 , u(1.,θ) ~ 1.202]
 
 # Space and time domains
 domains = [t ∈ IntervalDomain(0.0,1.0)]
-
 # Discretization
-dx = 0.1
-discretization = NeuralPDE.PhysicsInformedNN(dx)
-
-# Neural network and optimizer
-opt = Flux.ADAM(0.1)
+dt = 0.1
+# Neural network
 chain = FastChain(FastDense(1,12,Flux.σ),FastDense(12,1))
 
+discretization = NeuralPDE.PhysicsInformedNN(dt,
+                                             chain,
+                                             init_params = nothing;
+                                             phi = nothing,
+                                             autodiff=false,
+                                             derivative = nothing,
+                                             training_strategies = NeuralPDE.TrainingStrategies(stochastic_loss=false))
+
 pde_system = PDESystem(eq,bcs,domains,[t],[u])
 prob = NeuralPDE.discretize(pde_system,discretization)
-alg = NeuralPDE.NNDE(chain,opt,autodiff=false)
-phi,res = solve(prob,alg,verbose=true, maxiters=1000)
+
+initθ = discretization.initθ
+p = DiffEqBase.NullParameters()
+loss = (θ) -> prob.f(θ,p)
+
+res1 = DiffEqFlux.sciml_train(loss, initθ, Flux.ADAM(0.1); cb = cb, maxiters=1000)
+res2 = GalacticOptim.solve(prob,NelderMead(); cb = cb, maxiters=10000)
```

Comment:
```suggestion
res2 = GalacticOptim.solve(prob,BFGS(); cb = cb, maxiters=10000)
```

Reply:
@Vaibhavdixit02 @ChrisRackauckas BFGS() optimizer doesn't work correctly with Stochastic Loss. 
https://github.com/SciML/NeuralPDE.jl/blob/84c7982cf7fea1be94967d01b42c28c835002dd2/src/pinns_pde_solve.jl#L389

Reply:
That's expected. BFGS assumes the loss function is deterministic, so you need to use an optimizer robust to stochastic loss functions for that.

Reply:
ok, thanks. ADAM is one of them but not supported on GalacticOptim yet.

Reply:
we need that ASAP @Vaibhavdixit02